qsc/init_axis.py

```diff
@@ -7,10 +7,27 @@
 import logging
 from .spectral_diff_matrix import spectral_diff_matrix
 from .util import fourier_minimum
+from scipy.interpolate import CubicSpline as spline
 
 #logging.basicConfig(level=logging.DEBUG)
 logger = logging.getLogger(__name__)
 
+# Define periodic spline interpolants used in several scripts and plotting
+def R0_func(self,phi): return sum([self.rc[i]*np.cos(i*self.nfp*phi) for i in range(len(self.rc))])
+def Z0_func(self,phi): return sum([self.zs[i]*np.sin(i*self.nfp*phi) for i in range(len(self.zs))])
+def convert_to_spline(self,phi,array):
+    sp=spline(np.append(self.phi,2*np.pi/self.nfp), np.append(array,array[0]), bc_type='periodic')
+    return sp(np.mod(phi,2*np.pi/self.nfp))
```

Comment:
This approach to these splines is inefficient I think, which may be slowing down `Fourier_to_cylindrical` significantly. The inefficiency is that presently, a new spline is formed every time you want to evaluate a new phi. Forming the spline has some overhead. It is better to form the spline once, then call it many times for different phi. Instead of having `convert_to_spline` return the *result* of the spline evaluated at `phi`, just return the spline object `sp`, and then `sp(phi)` can be evaluated later. Then `phi` would be removed as an argument to `convert_to_splines`. The `np.mod` is unnecessary since the spline already handles periodicity.

Reply:
Thanks for pointing this out! It sped up the calculation from 20 seconds down to 5 seconds on the Section 5.1 Order r1 example.